include/seastar/rpc/rpc.hh

```diff
@@ -96,7 +97,43 @@ struct resource_limits {
     size_t max_memory = rpc_semaphore::max_counter(); ///< Maximum amount of memory that may be consumed by all requests
     /// Configures isolation for a connection based on its isolation cookie. May throw,
     /// in which case the connection will be terminated.
-    std::function<isolation_config (sstring isolation_cookie)> isolate_connection = default_isolate_connection;
+    class isolate_connection_wrapper {
+        using syncronous_isolation_function = std::function<isolation_config (sstring isolation_cookie)>;
+        using asyncronous_isolation_function = std::function<future<isolation_config> (sstring isolation_cookie)>;
+        using isolation_function_alternatives = std::variant<syncronous_isolation_function, asyncronous_isolation_function>;
+        isolation_function_alternatives _func;
+        struct isolation_function_visitor {
+                  isolation_function_visitor(const sstring& isolation_cookie) :
+                        _isolation_cookie(isolation_cookie) { }
+                  future<isolation_config> operator() (syncronous_isolation_function f) const {
+                      return futurize_invoke(f, _isolation_cookie);
+                  }
+                  future<isolation_config> operator() (asyncronous_isolation_function f) const {
+                      return f(_isolation_cookie);
+                  }
+              private:
+                  sstring _isolation_cookie;
+              };
```

Comment:
Wild indentation. Also, please move to around point of use, no need to clutter the interface with the visitor which isn't interesting to anybody.

Reply:
Ok, I will move it back to the way it was before. Meaning use and define the visitor in the point of use.